promises.cf.in

```diff
@@ -101,6 +101,12 @@ body common control
     (redhat|centos|suse|sles|opensuse|amazon_linux).cfe_yum_package_module_supported.!disable_inventory_package_refresh::
         package_inventory => { $(package_module_knowledge.platform_default) };
 
+    aix.!disable_inventory_package_refresh::
+      package_inventory => { $(package_module_knowledge.platform_default) };
```

Comment:
Would you not want to use `nimclient` AND `yum` for package inventory, or do they both result in the same, non-overlapping information?

Reply:
yes using nimclient and yum together would be way better. i think checking if "/usr/bin/yum" exists and setting some vars and classes (def.cf?), and and using this to configure ""platform_default", package_module and package_inventory . another benefit would be that no augments entry is needed.

pseudo code:

def.cf or something of the kind
```
classes:
          "yum_aix_exists" expression => fileexists("/usr/bin/yum");
```

packages.cf (not sure if this whole concern works with slist/ multiple values especially this -> "package_inventory => { $(package_module_knowledge.platform_default) }")
```
aix.!yum_aix_exists::
    "platform_default" string => "nimclient";

aix.yum_aix_exists::
    "platform_default" slist => { "nimclient", "yum" };
                  
```

Reply:
tested with this in lib/packages.cf, no changes in other files.

```
 aix::
      #"platform_default" -> { "CFE-3612" }
      #  comment => "set default package module via augments on aix else keep the default",
      #  string => ifelse( isvariable("def.packages_platform_default_alternate_aix"), "$(def.packages_platform_default_alternate_aix)",
      #                    "nimclient");

      "platform_default" slist => { "nimclient", "yum" };

```

error:

R: package_inventory is set to nimclient
R: package_inventory is set to yum
R: autorun is executing
R: default:packages_version_latest_warn_classes: found bundle default:packages_version_latest_warn_classes
R: DEBUG cfe_internal_management: Should actuate cfe_internal_core_main
   error: Can not find body for package module: $(package_module_knowledge.platform_default)

Reply:
Right, there are two things here, currently they are a bit conflated.

There is a default (singular, like the highlander there can be only one) package manager for a platform. That is the package manager that should be used by default if you say hey this software should be installed.

```
packages:
  "software" policy => "present";
```

Then, there is a list of package managers that should be used by default for _inventory_.
Currently, this is set to just the singular default package manager, we need to alter the implementation so that the list of package managers used for inventory by default includes the default package manager + whatever additional ones.